src/System.Runtime.Loader/tests/AssemblyLoadContextTests.netstandard1.7.cs

```diff
@@ -0,0 +1,329 @@
+                Assert.NotEqual(type.Assembly, asm);
+            });
+            alc.Unload();
+
+            checker.GcAndCheck();
+        }
+
+        [Fact]
+        public static void Unload_TwoCollectibleWithOneAssemblyAndOneInstanceReferencingAnother()
+        {
+            // We create 2 collectible ALC, load one assembly in each, create one instance in each, reference one instance from ALC1 to ALC2
+            // unload ALC2 -> check that instance 2 is there
+            // unload ALC1 -> we should receive 2 unload and instance should be gone
+            // We should receive the unloading event
+
+            var checker = new CollectibleChecker(2);
+            object instance1 = null;
+            var alc1 = TryCollectibleWithOneAssemblyLoaded(checker, (context, type) =>
+            {
+                instance1 = Activator.CreateInstance(type);
+            });
+
+            var alc2 = TryCollectibleWithOneAssemblyLoaded(checker, (context, type) =>
+            {
+                var instance2 = Activator.CreateInstance(type);
+                var field = instance1.GetType().GetField("Instance");
+                Assert.NotNull(field);
+
+                field.SetValue(instance1, instance2);
+            });
+
+            alc2.Unload();
+
+            // No assembly should be unloaded as alc1 reference alc2
+            checker.GcAndCheck(0, 1);
+
+            // Release instance1
+            instance1 = null;
+
+            // This should release instance2
+            checker.GcAndCheck(1);
+
+            // Finally release alc1
+            alc1.Unload();
+
+            // Check that we have all the events
+            checker.GcAndCheck();
+        }
+
+        private static AssemblyLoadContext TryCollectibleWithOneAssemblyLoaded(CollectibleChecker checker, Action<ResourceAssemblyLoadContext, Type> process = null)
+        {
+            var asmName = new AssemblyName(TestAssembly);
+            var alc = new ResourceAssemblyLoadContext(true) { LoadBy = LoadBy.Path };
+            alc.Unloading += context =>
+            {
+                checker.NotifyUnload();
+            };
+
+            var asm = alc.LoadFromAssemblyName(asmName);
+
+            Assert.NotNull(asm);
+            var type = asm.DefinedTypes.FirstOrDefault(t => t.Name == "TestClass");
+            Assert.NotNull(type);
+
+            if (process != null)
+            {
+                process(alc, type);
+            }
+
+            return alc;
+        }
+
+        private class CollectibleChecker
+        {
+            private const int MaxGCRetry = 10;
+            private const int MaxWaitTimePerGCRetryInMilli = 10;
+
+            private readonly int expectedCount;
+            private readonly ManualResetEvent evt;
+            private int unloadCount;
+
+            public CollectibleChecker(int expectedCount)
+            {
+                this.expectedCount = expectedCount;
```

Comment:
nit: We don't use `this.` for field accesses; use `_expectedCount = expectedCount;` instead.